src/censored.jl

```diff
@@ -0,0 +1,409 @@
+following methods should be implemented:
+- `censored(d::D, l::T, u::T) where {T <: Real}`
+- `censored(d::D, ::Missing, u::Real)`
+- `censored(d::D, l::Real, ::Missing)`
+"""
+censored
+function censored(d::UnivariateDistribution, l::T, u::T) where {T<:Real}
+    return Censored(d, l, u)
+end
+function censored(d::UnivariateDistribution, ::Missing, u::Real)
+    return Censored(d, missing, u)
+end
+function censored(d::UnivariateDistribution, l::Real, ::Missing)
+    return Censored(d, l, missing)
+end
+censored(d::UnivariateDistribution, l::Real, u::Real) = censored(d, promote(l, u)...)
+censored(d::UnivariateDistribution, ::Missing, ::Missing) = d
+
+"""
+    Censored
+
+Generic wrapper for a [`censored`](@ref) distribution.
+"""
+struct Censored{
+    D<:UnivariateDistribution,
+    S<:ValueSupport,
+    T<:Real,
+    TL<:Union{T,Missing},
+    TU<:Union{T,Missing},
+} <: UnivariateDistribution{S}
+    uncensored::D      # the original distribution (uncensored)
+    lower::TL      # lower bound
+    upper::TU      # upper bound
+    function Censored(d::UnivariateDistribution, l::T, u::T) where {T<:Real}
+        _is_non_empty_interval(l, u) ||
+            error("the lower bound must be less than or equal to the upper bound")
+        new{typeof(d), value_support(typeof(d)), T, T, T}(d, l, u)
+    end
+    function Censored(d::UnivariateDistribution, l::Missing, u::T) where {T <: Real}
+        new{typeof(d), value_support(typeof(d)), T, Missing, T}(d, l, u)
+    end
+    function Censored(d::UnivariateDistribution, l::T, u::Missing) where {T <: Real}
+        new{typeof(d), value_support(typeof(d)), T, T, Missing}(d, l, u)
+    end
+end
+
+function censored(d::Censored, l::T, u::T) where {T<:Real}
+    return censored(
+        d.uncensored,
+        d.lower === missing ? l : max(l, d.lower),
+        d.upper === missing ? u : min(u, d.upper),
+    )
+end
+function censored(d::Censored, ::Missing, u::Real)
+    return censored(d.uncensored, d.lower, d.upper === missing ? u : min(u, d.upper))
+end
+function censored(d::Censored, l::Real, ::Missing)
+    return censored(d.uncensored, d.lower === missing ? l : max(l, d.lower), d.upper)
+end
+
+function params(d::Censored)
+    d0params = params(d.uncensored)
+    return if d.lower === missing
+        (d0params..., d.upper)
+    elseif d.upper === missing
+        (d0params..., d.lower)
+    else
+        (d0params..., d.lower, d.upper)
+    end
+end
+
+partype(d::Censored) = partype(d.uncensored)
+
+Base.eltype(::Type{<:Censored{D,S,T}}) where {D,S,T} = promote_type(T, eltype(D))
+
+#### Range and Support
+
+function islowerbounded(d::Censored)
+    return (
+        islowerbounded(d.uncensored) ||
+        (d.lower !== missing && cdf(d.uncensored, d.lower) > 0)
+    )
+end
+function isupperbounded(d::Censored)
+    return (
+        isupperbounded(d.uncensored) ||
+        (d.upper !== missing && _ccdf_inc(d.uncensored, d.upper) > 0)
```

Comment:
A bit difficult to read since `_ccdf_inc` is only defined at the end of this file.

Reply:
It was not obvious also what `_ccdf_inc` stands for, so maybe a more descriptive name would be helpful.

Reply:
Renaming to `_ccdf_inclusive`